include/zephyr/net/icmp.h

```diff
@@ -191,6 +191,30 @@ int net_icmp_send_echo_request(struct net_icmp_ctx *ctx,
 			       struct net_icmp_ping_params *params,
 			       void *user_data);
 
+/**
+ * @brief Send ICMP echo request message from k_work handler.
+ *
+ * @details This function can be used to send ICMP echo request from
+ *          a k_work handler. This variant will not wait if the network
```

Comment:
Technically this is only true for system workqueue, blocking on a custom workqueue should have no side effects (warnings printed).